api/envoy/config/route/v3/route_components.proto

```diff
@@ -1564,3 +1571,35 @@ message QueryParameterMatcher {
     bool present_match = 6;
   }
 }
+
+// HTTP Internal Redirect :ref:`architecture overview <arch_overview_internal_redirects>`.
+message InternalRedirectPolicy {
+  message Predicate {
+    google.protobuf.Any typed_config = 1;
+  }
```

Comment:
nit: I don't think you need this wrapper message anymore? You could just have a repeated any below with name `predicate_type_configs` or something like that? Up to you.

Reply:
Done.